### PR TITLE
Easier access to logs for the web demo

### PR DESCRIPTION
**Pull request by yihozhang**
After this PR:
image
image

### CODE REVIEW

File: web-demo/static/worker.js [hunk 1]
```diff
@@ -8,16 +8,22 @@ async function work() {
...
-            self.postMessage({dot: "", text: "Something panicked! Check the console logs..."});
+            self.postMessage({ dot: "", text: "Something panicked! Check the console logs...", log: logbuffer });
         }
     };
 }
 
+logbuffer = [];
+function log(level, str) {
+    logbuffer.push([level, str]);
+}
```
Comment (saulshanabrook) [inline web-demo/static/worker.js:27]: Is this used anywhere else?

**Review by saulshanabrook — APPROVED** [review]
Nice! I just tried it and it seems to work well.

The frontend code is starting to get more complicated and it might make more sense to just send structured data from the backend instead of HTML and style it in the frontend, but it seems fine for now.

I imagine if the editor keeps getting more complicated it might be worth moving to something like React in the future to manage state/complexity.

File: web-demo/static/worker.js (hunk 1 above)
Comment (yihozhang) [reply web-demo/static/worker.js:27]: This is used by the Rust WebDemoLogger!

**Comment by yihozhang** [comment]
Yeah, I agree the frontend code is getting more complex. It makes sense to switch to a more established framework for managing states, etc.

**Pull request merged**